20161223 PR

hans-maws opened the pull request:
--> Added a **validateSignatureForGet** and a **validateSignatureForPost** method into AuthyUtil component to help the user validate the signature of the callback for OneTouch without too much problems or need to understand encryption.
--> api_key and api_url are now loaded from environment variables or properties file with **loadProperties** method from **AuthyUtil** (*environment variables will override properties file*)
--> Fixed a bad issue were parameters were not being properly set by the Resource.prepareGet method so options like force=true were never sent.
--> Updated README.md file with one touch information.
--> Added some extra Unit testing cases./
--> Fixed from String comparison issues (using == instead of .equals for Strings)
--> Did an initial code refactor.
--> Updated existent JUnit test cases that were failing with the updated messages.

hans-maws committed:
--> Fixed a bad issue were parameters were not being properly set by the Resource.prepareGet method so options like force=true were never sent.
--> Updated README.md file with one touch information.
--> Added some extra Unit testing cases./
--> Fixed from String comparison issues (using == instead of .equals for Strings)
--> Did an initial code refactor.

hans-maws committed:
--> Added a validateSignature method into AuthyUtil component to help the user validate the signature of the callback for OneTouch without too much problems.
--> api_key and api_url are now loaded from environment variables or properties file with loadProperties method from AuthyUtil (environment variables will override properties file)
--> Fixed a bad issue were parameters were not being properly set by the Resource.prepareGet method so options like force=true were never sent.
--> Updated README.md file with one touch information.
--> Added some extra Unit testing cases./
--> Fixed from String comparison issues (using == instead of .equals for Strings)
--> Did an initial code refactor.
--> Updated existent JUnit test cases that were failing with the updated messages.

hans-maws committed:
--> Added a validateSignatureForGet and a validateSignatureForPost method into AuthyUtil component to help the user validate the signature of the callback for OneTouch without too much problems or need to understand encryption.
--> api_key and api_url are now loaded from environment variables or properties file with loadProperties method from AuthyUtil (environment variables will override properties file)
--> Fixed a bad issue were parameters were not being properly set by the Resource.prepareGet method so options like force=true were never sent.
--> Updated README.md file with one touch information.
--> Added some extra Unit testing cases./
--> Fixed from String comparison issues (using == instead of .equals for Strings)
--> Did an initial code refactor.
--> Updated existent JUnit test cases that were failing with the updated messages.

vargasx commented on src/test/resources/com/authy/api/test.properties:
```diff
@@ -1,7 +1,9 @@
-onetouch_udid=abcf88ab0-a8af-0134-087f-0ae83edadd62
+user_id=28894864
+#this will be just an UDID user for the unit test, it will be not valid for your app.
+onetouch_udid=ee3aff60-a8f9-0134-aed0-0a5fb1fd6492
```
please find all udid typos ?

hans-maws replied:
will do

vargasx commented on src/test/java/com/authy/api/TestPhoneVerification.java:
```diff
@@ -44,18 +33,18 @@ public void itTestsVerificationStartEs() {
+        Verification result = client.start("+1 775-461-5609", "1", "sms", params);
 
-        Assert.assertEquals("Text message sent to +1 555-555-5555.", result.getMessage());
+        Assert.assertEquals("Text message sent to +1 775-461-5609.", result.getMessage());
```
can we move country codes and phone numbers to properties as well ?

what do you think?

hans-maws replied:
done

vargasx commented on src/main/java/com/authy/AuthyUtil.java:
```diff
@@ -0,0 +1,196 @@
+     * @return true if the signature ios valid, false otherwise
+     * @throws UnsupportedEncodingException if the string parameters have problems with UTF-8 encoding.
+     */
+    public static boolean validateSignatureForPost(String body, Map<String, String> headers, String url, String authyToken) throws UnsupportedEncodingException {
```
could you add unit test for this ?

vargasx commented on src/main/java/com/authy/AuthyUtil.java:
```diff
@@ -0,0 +1,196 @@
+     * @return true if the signature ios valid, false otherwise
+     * @throws UnsupportedEncodingException if the string parameters have problems with UTF-8 encoding.
+     */
+    public static boolean validateSignatureForGet(Map<String, String> params, Map<String, String> headers, String url, String authyToken) throws UnsupportedEncodingException {
```
could you add unit test for this ?

vargasx commented on README.md:
```diff
@@ -179,6 +179,80 @@ If the request was successful, you will need to store the authy id in your datab
+  for example, using the 
+  'sendApprovalRequest' method you will create an ApprovalRequest
+  in OneTouch, then the approval request
+  status can be queried by using the  `uuid` provided. (NOTE: The UDID is the unique identifier for the request sent to OneTouch)    
```
udid typo?

hans-maws replied:
on it

vargasx commented on README.md:
````diff
@@ -179,6 +179,80 @@ If the request was successful, you will need to store the authy id in your datab
+
+  If you want to query the ApprovalRequest status
+```java
+  String status = client.getOneTouch().getApprovalRequestStatus(udid).getStatus();
````
udid typo?

hans-maws replied:
on it

vargasx commented on README.md:
````diff
@@ -179,6 +179,80 @@ If the request was successful, you will need to store the authy id in your datab
+When you make a OneTouch request, you will get a OneTouchResponse object which in turn will have a getApprovalRequest() accessor to retrieve the metadata from the given request, for example:
+
+```java
+    OneTouchResponse response = client.getOneTouch().getApprovalRequestStatus(udid);
````
udid typo?

hans-maws replied:
on it

vargasx commented on README.md:
```diff
@@ -179,6 +179,80 @@ If the request was successful, you will need to store the authy id in your datab
+    // was the user notified?
+    System.out.println("The user was "+(approvalRequest.isNotified()?"":"not")+" notified.");
+    // What is the status of the request?
+    System.iout.print("The status of the request is: "+approvalRequest.getStatus());
```
.iout typo?

hans-maws replied:
done

hans-maws committed:
--> Fixed UDID->UUID typos.
--> Added Unit testing for AuthyUtil signature helper that works with GET and POST
--> Added a validateSignatureForGet and a validateSignatureForPost method into AuthyUtil component to help the user validate the signature of the callback for OneTouch without too much problems or need to understand encryption.
--> api_key and api_url are now loaded from environment variables or properties file with loadProperties method from AuthyUtil (environment variables will override properties file)
--> Fixed a bad issue were parameters were not being properly set by the Resource.prepareGet method so options like force=true were never sent.
--> Updated README.md file with one touch information.
--> Added some extra Unit testing cases./
--> Fixed from String comparison issues (using == instead of .equals for Strings)
--> Did an initial code refactor.
--> Updated existent JUnit test cases that were failing with the updated messages.

vargasx commented on README.md:
```diff
@@ -179,6 +179,80 @@ If the request was successful, you will need to store the authy id in your datab
+
+    HashMap<String, String> logos = new HashMap<String, String>();
+
+    OneTouchResponse response = client.getOneTouch().sendApprovalRequest(Integer.parseInt(103), "Authorize OneTouch Unit Test", details, hidden, logos, 86400);
```
Could you please see this method signature from ruby `https://github.com/authy/authy-ruby/blob/master/lib/authy/onetouch.rb#L12` it only receives a hash. could we change it here to receive one hash and validate it contains authy_id, message, also could you check how we validate logos in Ruby lib in same file `https://github.com/authy/authy-ruby/blob/master/lib/authy/onetouch.rb`

vargasx replied:
@hans-maws lets wait before you make changes here base on suggestion. I'll review and make a further comment.

vargasx commented on README.md:
```diff
@@ -179,6 +179,80 @@ If the request was successful, you will need to store the authy id in your datab
+    OneTouchResponse response = client.getOneTouch().sendApprovalRequest(Integer.parseInt(103), "Authorize OneTouch Unit Test", details, hidden, logos, 86400);
+    // If the request was successfuly created.
+    assert response.isSuccess();
+    String uuid = response.getApprovalRequest().getUUID();
```
I'm trying to use this code in eclipse but I'm getting this message `The type OneTouchResponse.ApprovalRequest is no visible`

hans-maws committed:
Improvements

hans-maws committed:
Update README.md

hans-maws committed:
--> Fixed UDID->UUID typos.
--> Added Unit testing for AuthyUtil signature helper that works with GET and POST
--> Added a validateSignatureForGet and a validateSignatureForPost method into AuthyUtil component to help the user validate the signature of the callback for OneTouch without too much problems or need to understand encryption.
--> api_key and api_url are now loaded from environment variables or properties file with loadProperties method from AuthyUtil (environment variables will override properties file)
--> Fixed a bad issue were parameters were not being properly set by the Resource.prepareGet method so options like force=true were never sent.
--> Updated README.md file with one touch information.
--> Added some extra Unit testing cases./
--> Fixed from String comparison issues (using == instead of .equals for Strings)
--> Did an initial code refactor.
--> Updated existent JUnit test cases that were failing with the updated messages.

hans-maws committed:
Merge branch 'onetouch' of hans-maws:MobileAWS/authy-java into onetouch

hans-maws committed:
--> Improved exception handling for OneTouch
--> Added new Unit tests for OneTouch
--> Code cleanup
--> Better documentation
--> Fixed some typos.

hans-maws committed:
--> Updated Readme file.
--> Improved exception handling for OneTouch
--> Added new Unit tests for OneTouch
--> Code cleanup
--> Better documentation
--> Fixed some typos.

hans-maws committed:
--> Updated Readme file.
--> Improved exception handling for OneTouch
--> Added new Unit tests for OneTouch
--> Code cleanup
--> Better documentation
--> Fixed some typos.

hans-maws committed:
--> Updated Readme file.
--> Improved exception handling for OneTouch
--> Added new Unit tests for OneTouch
--> Code cleanup
--> Better documentation
--> Fixed some typos.
